anime_downloader/sites/animesvostfr.py

```diff
@@ -0,0 +1,247 @@
+        
+        link_stream_url = "https://www1.animesvostfr.net/ajax-get-link-stream/"
+        
+        SERVERS = [
+            'rapidvideo',
+            'photoss'
+        ]
+
+        def _get_sources(self):
+            
+            #version = self.config['version'] 
+            #server = self.config['server']
+            #fallback = self.config['fallback_servers']
+            
+            logger.debug( "_get_sources url: %s", self.url )
+            
+            #https://www1.animesvostfr.net/ajax-get-link-stream/?server=rapidvideo&filmId=45532
+            #-> https://there.to/v/-zln1apmp3g423r
+            
+            filmid = None
+            episodeid = None
+            episodenum = None
+            
+            soup = helpers.soupify(helpers.get(self.url, params={'server' : 'rapidvideo'}, cf=True))
+            
+            shortlink = soup.find( 'link', {'rel': 'shortlink'})
+            logger.debug( "shortlink: {}".format(shortlink) )
+            if shortlink:
+                #logger.debug( "shortlink2: {}".format(shortlink['href']) )
+                ids = re.findall(r'.*/?p=([^"]+)', shortlink['href'])
+                #logger.debug( "shortlink2: {}".format(ids) )
+                if ids and len(ids) > 0:
+                    filmid = ids[0]
+                
+                
+            
+
+            select = soup.select_one('div.list-episodes  select')
+            option = None
+            
+            if select:
+                option = select.find('option', {'selected': True})
+            if option:
+                episodeid = option['episodeid']
+                episodenum = option.text
+            
+            logger.debug( "_get_sources  filmid: %s episodeid: %s episodenum: %s", str(filmid), str(episodeid), str(episodenum) )
+            
+            
+            if not filmid:
+                filmid = episodeid
+            
+            episodeurl = None
+            if filmid:
+                episodeurl = helpers.get(self.link_stream_url, params={'server' : 'rapidvideo', 'filmId' : filmid}, cf=True).text.strip()
+                
+                if not episodeurl or episodeurl == '':
+                    episodeurl = helpers.get(self.link_stream_url, params={'server' : 'photoss', 'filmId' : filmid}, cf=True).text.strip()
+                    
+            else:
+                raise NotFoundError
+                
+            logger.debug( "_get_sources url:%s", episodeurl )
+            
+            
+            
+            
+            #https://youtubedownloader.cx/v/y0661ue6-8jzj34
+            #https://youtubedownloader.cx/api/source/y0661ue6-8jzj34
+            
+            
+            #https://play.comedyshow.to/embedplay/4a2faf54b99e8d1152e598b24916d0d9
+            #<iframe src="https://www.fembed.net/v/1ekm5ij-3l2rp8p"
+            #https://feurl.com/api/source/1ekm5ij-3l2rp8p
+            # POST r: https://play.comedyshow.to
+            #d: feurl.com
```

Comment:
Not sure why you've commented this - since you don't seem to be handling fembed/feurl links at a glance, but they can be given to the gcloud extractor.

Reply:
OK i can use  the gcloud extractor but, do i need to add code for choose the good extractor, or that is automatic ?

Also, i looked the gcloud extractor, and this don't use fallback_qualities ?

If the self.quality is not found, this don't choose another quality in the prefered order of the fallback config ?